src/main/scala/org/dia/algorithms/mcc/GTGRunner.scala

```diff
@@ -348,11 +350,20 @@ class GTGRunner(val masterURL: String,
     val subgraphs = sc.sparkContext.parallelize(edgeListRDDIndexed.toSeq)
       .map(MCCOps.mapEdgesToBuckets(_, maxParitionSize, buckets))
       .groupByKey()
-    val subgraphsFound = MCCOps.findSubgraphsIteratively(subgraphs, 1, maxParitionSize, minGraphLength, sc.sparkContext)
+    val subgraphsFound = MCCOps.findSubgraphsIteratively(subgraphs, 1, maxParitionSize,
+      minGraphLength, sc.sparkContext)
     for(x <- subgraphsFound) {
-      print(x._2.toList)
+      logger.info("Edges remaning : " + x._2.toList)
     }
     /**
+     * Generate the netcdfs
+     */
+
+    val edgeTupleRDD = edgeListRDD.map({ x =>
+      MCSUtils.get_node_data(x, broadcastedNodeMap.value, lat, lon, false)})
+      .collect()
```

Comment:
@sujen1412 
Collecting is one way of getting the transformation to actually be evaluated on an RDD. However, we end up also incurring the cost of a "collect" even though nothing is really being communicated back. How about trying to use foreach()? 

There's another way to have side effects on RDD's without collecting, and that's through the foreach() function.
Take a look here.
https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/rdd/RDD.scala#L889-L895

We avoid the network cost of a collect.